S3Proxy/handlers.go

```diff
@@ -19,16 +20,13 @@ func StatusHandler(w http.ResponseWriter, req *http.Request) {
 
 // The default handler used for everything else
 func DefaultHandler(w http.ResponseWriter, req *http.Request) {
-	keyMap, err := S3ValidateKey(req.URL.Path[1:])
+	params := req.URL.Query()
+	parts := strings.SplitN(req.URL.Path[1:], "/", 2)
+	bucket, key := parts[0], parts[1]
+	obj, err := S3GetObject(bucket, key, params.Get("aws_region"))
 	if err != nil {
-		http.Error(w, "404: Not Found", 404)
-		return
+		http.Error(w, err.Message, err.Code)
```

Comment:
Dammit seems like I missed the return after cleaning up. Will resubmit with the return included.